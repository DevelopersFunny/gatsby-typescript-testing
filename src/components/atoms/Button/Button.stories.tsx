import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import { Button } from '.'

export default {
  title: 'Atoms/Button',
  component: Button,
  argTypes: {
    backgroundColor: { control: 'color' },
    color: { control: 'color' },
  },
} as ComponentMeta<typeof Button>

const Template: ComponentStory<typeof Button> = args => <Button {...args} />

export const Primary = Template.bind({})
Primary.args = {
  label: 'Primary Button',
  variant: 'primary',
}
Primary.parameters = {
  jest: ['button.test.tsx'],
}

export const Secondary = Template.bind({})
Secondary.args = {
  label: 'Secondary Button',
  variant: 'secondary',
}
Secondary.parameters = {
  jest: ['button.test.tsx'],
}

export const Small = Template.bind({})
Small.args = {
  label: 'Small Button',
  size: 'small',
  variant: 'primary',
}
Small.parameters = {
  jest: ['button.test.tsx'],
}

export const Large = Template.bind({})
Large.args = {
  label: 'Large Button',
  size: 'large',
  variant: 'primary',
}
Large.parameters = {
  jest: ['button.test.tsx'],
}

export const CustomBackgroundColor = Template.bind({})
CustomBackgroundColor.args = {
  backgroundColor: 'red',
  label: 'Custom Color',
  variant: 'primary',
}
CustomBackgroundColor.parameters = {
  jest: ['button.test.tsx'],
}

export const CustomTextColor = Template.bind({})
CustomTextColor.args = {
  color: 'gray',
  label: 'Custom Color',
  variant: 'primary',
}
CustomTextColor.parameters = {
  jest: ['button.test.tsx'],
}
