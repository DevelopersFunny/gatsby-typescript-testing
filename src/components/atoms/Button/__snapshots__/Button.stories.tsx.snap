// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories storyshots Atoms/Button Button Click 1`] = `
<button
  className="css-1mkuulw"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  ClickEv Button
</button>
`;

exports[`Stories storyshots Atoms/Button Contained 1`] = `
<button
  className="css-1mkuulw"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Contained Button
</button>
`;

exports[`Stories storyshots Atoms/Button Custom Background Color 1`] = `
<button
  className="css-1bhnfi6"
  style={
    Object {
      "backgroundColor": "#ee6d6d",
      "color": undefined,
    }
  }
>
  Custom Color
</button>
`;

exports[`Stories storyshots Atoms/Button Custom Text Color 1`] = `
<button
  className="css-1bhnfi6"
  style={
    Object {
      "backgroundColor": undefined,
      "color": "green",
    }
  }
>
  Custom Color
</button>
`;

exports[`Stories storyshots Atoms/Button Error 1`] = `
<button
  className="error css-1bhnfi6"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Error Button
</button>
`;

exports[`Stories storyshots Atoms/Button Large 1`] = `
<button
  className="css-okjmew"
  size="large"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Large Button
</button>
`;

exports[`Stories storyshots Atoms/Button Normal 1`] = `
<button
  className="css-1mkuulw"
  size="normal"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Normal Button
</button>
`;

exports[`Stories storyshots Atoms/Button Outlined 1`] = `
<button
  className="css-13j8qy9"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Outlined Button
</button>
`;

exports[`Stories storyshots Atoms/Button Secondary 1`] = `
<button
  className="secondary css-1bhnfi6"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Secondary Button
</button>
`;

exports[`Stories storyshots Atoms/Button Small 1`] = `
<button
  className="css-dhkxoc"
  size="small"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Small Button
</button>
`;

exports[`Stories storyshots Atoms/Button Success 1`] = `
<button
  className="success css-1bhnfi6"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Success Button
</button>
`;

exports[`Stories storyshots Atoms/Button Text 1`] = `
<button
  className="css-1bhnfi6"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Text Button
</button>
`;

exports[`Stories storyshots Atoms/Button Warning 1`] = `
<button
  className="warning css-1bhnfi6"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  Warning Button
</button>
`;
